### PR TITLE
Rename no-master-block setting

### PR DESCRIPTION
**Pull request by DaveCTurner**
Replaces `discovery.zen.no_master_block` with `cluster.no_master_block`. Any
value set for the old setting is now ignored.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java [hunk 1]
```diff
@@ -103,7 +102,7 @@
     private final JoinHelper joinHelper;
     private final NodeRemovalClusterStateTaskExecutor nodeRemovalExecutor;
     private final Supplier<CoordinationState.PersistedState> persistedStateSupplier;
-    private final DiscoverySettings discoverySettings;
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java]: 🕺

**Comment by DaveCTurner** [comment]
FAO @gwbrown as this is a breaking change. We explored a number of options for renaming this dynamic setting so as to fall back to the legacy setting name but all had drawbacks. Since this setting is very rarely adjusted, and only takes effect when the cluster is broken, we opted simply to start ignoring it in favour of the new name.

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by gwbrown** [comment]
Thanks for the ping @DaveCTurner, I've added this to the list at https://github.com/elastic/elasticsearch/issues/36024. This looks like a setting that would typically be set via the cluster settings API (rather than elasticsearch.yml), correct?

**Comment by DaveCTurner** [comment]
> This looks like a setting that would typically be set via the cluster settings API (rather than elasticsearch.yml), correct?

TBH the overwhelmingly typical case is that this setting isn't set at all. I don't think we can say whether it's more or less likely to be set as a cluster setting vs a node setting.

**Pull request merged**